mava/components/jax/training/step.py

```diff
@@ -113,6 +137,41 @@ def on_training_step(self, trainer: SystemTrainer) -> None:
         trainer.store.trainer_logger.write({**results})
 
 
+class Step(Component):
+    @abc.abstractmethod
+    def on_training_step_fn(self, trainer: SystemTrainer) -> None:
+        """[summary]"""
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "sgd_step"
+
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        TrainerDataset required for config sample_batch_size.
+        ParallelSequenceAdder required for config sequence_length.
+        BaseTrainerInit required to set up trainer.store.networks
+        and trainer.store.trainer_agent_net_keys.
+        Networks required to set up trainer.store.key.
+
+        Returns:
+            List of required component classes.
+        """
+        return [
+            TrainerDataset,
+            mava.components.jax.building.adders.ParallelSequenceAdder,
```

Comment:
Step should just depend on Adders as per my first comment on abstract components